src/framework/theme/components/toggle/toggle.component.ts

```diff
@@ -0,0 +1,339 @@
+
+  onChange: any = () => { };
+  onTouched: any = () => { };
+
+  private direction: NbLayoutDirection;
+  private destroy$: Subject<void> = new Subject<void>();
+
+  /**
+   * Toggle checked
+   * @type {boolean}
+   * @private
+   */
+  @Input()
+  get checked(): boolean {
+    return this._checked;
+  }
+  set checked(value: boolean) {
+    this._checked = value;
+  }
+  private _checked: boolean = false;
+
+  /**
+   * Controls input disabled state
+   */
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = convertToBoolProperty(value);
+  }
+  private _disabled: boolean = false;
+
+  /**
+   * Toggle status.
+   * Possible values are: `primary` (default), `success`, `warning`, `danger`, `info`
+   */
+  @Input()
+  status: '' | NbComponentStatus = '';
+
+  /**
+   * Controls toggle aria checked
+   */
+  @Input()
+  get ariaChecked(): boolean {
+    return this._ariaChecked;
+  }
+  set ariaChecked(value: boolean) {
+    this._ariaChecked = convertToBoolProperty(value);
+  }
+  private _ariaChecked: boolean = false;
+
+  /**
+   * Toggle label position.
+   * Possible values are: `left`, `right`, `start`, `end` (default)
+   */
+  @Input() labelPosition: 'left' | 'right' | 'start' | 'end' = 'end';
+
+  /**
+   * Output when checked state is changed by a user
+   * @type EventEmitter<boolean>
+   */
+  @Output() checkedChange = new EventEmitter<boolean>();
+
+  @HostBinding('class.status-primary')
+  get primary() {
+    return this.status === 'primary';
+  }
+
+  @HostBinding('class.status-success')
+  get success() {
+    return this.status === 'success';
+  }
+
+  @HostBinding('class.status-warning')
+  get warning() {
+    return this.status === 'warning';
+  }
+
+  @HostBinding('class.status-danger')
+  get danger() {
+    return this.status === 'danger';
+  }
+
+  @HostBinding('class.status-info')
+  get info() {
+    return this.status === 'info';
+  }
+
+  @HostBinding('class.toggle-label-left')
+  get labelLeft() {
+    return this.labelPosition === 'left';
+  }
+
+  @HostBinding('class.toggle-label-right')
+  get labelRight() {
+    return this.labelPosition === 'right';
+  }
+
+  @HostBinding('class.toggle-label-start')
+  get labelStart() {
+    return this.labelPosition === 'start';
+  }
+
+  @HostBinding('class.toggle-label-end')
+  get labelEnd() {
+    return this.labelPosition === 'end';
+  }
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private layoutDirection: NbLayoutDirectionService,
+  ) {}
+
+  ngOnInit(): void {
+    this.layoutDirection.onDirectionChange()
+      .pipe(takeUntil(this.destroy$))
+      .subscribe(() => {
+      this.direction = this.layoutDirection.getDirection();
+    });
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.complete();
+  }
+
+  checkState(): string {
+    if (this.checked) {
+      if (this.direction === NbLayoutDirection.LTR) {
```

Comment:
If it's the only place where we need to know the direction, we can use `this.directionService.getDirection()` method to get current direction. And then subscription from `ngOnInit` is unneeded.